scripts/source-maintenance.sh

```diff
@@ -413,4 +457,4 @@ cat scripts/boilerplate.h doc/debug-sections.tmp2 >doc/debug-sections.txt
 rm doc/debug-sections.tmp doc/debug-sections.tmp2
 rm -f boilerplate_fix.sed
 
-exit $SeenErrors
+exit ${SeenErrors}
```

Comment:
IMO, this out-of-scope change and similar changes throughout the code should be undone. I am not sure they are desirable (their merit should be discussed) but even if they are desirable, they should not be applied to otherwise unmodified-by-the-PR code because such widespread changes make review unnecessary difficult.

Reply:
I will undo all lines where the adding the parentheses is the only change. If a line is touched in any other way, I'll leave this in. Is this an agreeable scope?

Reply:
> I will undo all lines where the adding the parentheses is the only change. If a line is touched in any other way, I'll leave this in. Is this an agreeable scope?

I am not sure, but I think you are implying that we should be moving (in general) towards using `{}` decorations for all shell variables, replacing every `$x` with `${x}`. Is that what you think we should be doing? If yes, why? I have not studied the issue yet, but on the surface, `$x` is better than `${x}` in all contexts where there is no room for misinterpreting the variable name boundaries (which is the vast majority of contexts!).

Let's clarify what the end goal is before deciding how to get there.